smslibrary/src/test/java/com/eis/smslibrary/SMSManagerTest.java

```diff
@@ -34,13 +45,12 @@ public void singletonInstance() {
 
     @Test()
     public void validMessage_isSent() {
-        SMSCore.setManager(managerMock);
-        instance.sendMessage(VALID_MESSAGE);
+        smsManager.sendMessage(VALID_MESSAGE);
         verify(managerMock).sendTextMessage(PEER_TEXT, null, SENT_TEXT, null, null);
     }
 
     @Test(expected = NullPointerException.class)
     public void nullMessage_throws() {
-        instance.sendMessage(null);
+        smsManager.sendMessage(null);
```

Comment:
Wait a second, I was wrong in https://github.com/Cogno-IDU/smslibrary/pull/39#discussion_r362356126, the `NullPointerException` is thrown in `DefaultSMSMessageParseStrategy.parseData()`, because it calls `message.getData()` and `message` points to `null`, so method `setup()` in this test doesn't need to be executed before every test, you can move its body to method `validMessage_isSent()`, since it's the only one using mocks.

Reply:
Since it's already working I don't think this kind of edit is necessary. Also in general is not a particularly good practice, what if I add another test which needs setup() ? it obviously won't work unless I call it, but then I'll need to re-create `setup()`. So I think I'll leave it there

Reply:
Ok, I thought methods annotated with `@Before` had to contain only setup operations that need to be performed before every test in the class. In cases where a lot of tests need to be run and performance matters, it's probably better to use `@BeforeClass` for operations that need to be executed only once, before all tests are executed.